src/lambda/wizards/samDeployWizard.ts

```diff
@@ -382,6 +459,10 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
                 }
             }
         })
+
+        const val = picker.verifySinglePickerOutput(pickedItem)
+
+        return (val && val.label !== s3ErrorLabel && val.label !== s3NoBucketsLabel) ? val.label : undefined
```

Comment:
If you define a type like 
`type xxx = vscode.QuickPickItem & { isBucket: boolean }`
you can simplify this to not text match against specific labels.

Reply:
✔️